File: 03. Stack and Queue/02. Next Greater Element/main.js
function nextGreaterElement(nums) {
    let stack = [];
    let result = new Array(nums.length).fill(-1);

    for (let i = 0; i < nums.length; i++) {
        while (stack.length > 0 && nums[stack[stack.length - 1]] < nums[i]) {
            result[stack.pop()] = nums[i];
        }
        stack.push(i);
    }

    return result;
}

// Main function
function main() {
    let nums = [4, 5, 2, 10, 8];
    let result = nextGreaterElement(nums);

    console.log("Next Greater Elements:");
    console.log(result);
}

// Run the main function
main();

File: 03. Stack and Queue/02. Next Greater Element/main.py
def next_greater_element(nums):
    stack = []
    result = [-1] * len(nums)

    for i in range(len(nums)):
        while stack and nums[stack[-1]] < nums[i]:
            result[stack.pop()] = nums[i]
        stack.append(i)

    return result

# Main function
def main():
    nums = [4, 5, 2, 10, 8]
    result = next_greater_element(nums)

    print("Next Greater Elements:")
    print(result)

# Run the main function
if __name__ == "__main__":
    main()
